File: Examples/AspNetMvcVB/Controllers/DocumentViewerController.Display.vb
Imports GleamTech.AspNet.UI
Imports GleamTech.DocumentUltimate.AspNet.UI

Namespace Controllers
    Partial Public Class  DocumentViewerController
        Inherits Controller

        Public Function Display() As ActionResult

            Dim documentViewer1 = New DocumentViewer With {
                .Id = "documentViewer1",
                .Width = 800,
                .Height = 600,
                .Resizable = True,
                .Document = "~/App_Data/ExampleFiles/Default.pdf",
                .Hidden = True
            }

            Dim documentViewer2 = New DocumentViewer With {
                .Id = "documentViewer2",
                .Width = 800,
                .Height = 600,
                .Resizable = True,
                .Document = "~/App_Data/ExampleFiles/Default.pdf",
                .Hidden = True,
                .DisplayMode = DisplayMode.Window,
                .WindowOptions = New WindowOptions With {
                    .Title = "DocumentViewer as a modal dialog",
                    .Modal = True,
                    .Maximizable = True,
                    .Minimizable = True
                }
            }

            Dim documentViewer3 = New DocumentViewer With {
                .Id = "documentViewer3",
                .Width = 800,
                .Height = 600,
                .Resizable = True,
                .Document = "~/App_Data/ExampleFiles/Default.pdf",
                .Hidden = True,
                .DisplayMode = DisplayMode.Panel,
                .PanelOptions = New PanelOptions With {
                    .Title = "DocumentViewer as a panel",
                    .Collapsible = True
                }
            }

            Return View({documentViewer1, documentViewer2, documentViewer3})

        End Function

    End Class
End Namespace

File: Examples/AspNetMvcCS/Controllers/DocumentViewerController.Display.cs
using GleamTech.AspNet.UI;
using GleamTech.DocumentUltimate.AspNet.UI;
using System.Web.Mvc;

namespace GleamTech.DocumentUltimateExamples.AspNetMvcCS.Controllers
{
    public partial class DocumentViewerController
    {
        [AcceptVerbs(HttpVerbs.Get | HttpVerbs.Post)]
        public ActionResult Display()
        {
            var documentViewer1 = new DocumentViewer
            {
                Id = "documentViewer1",
                Width = 800,
                Height = 600,
                Resizable = true,
                Document = "~/App_Data/ExampleFiles/Default.pdf",
                Hidden = true
            };

            var documentViewer2 = new DocumentViewer
            {
                Id = "documentViewer2",
                Width = 800,
                Height = 600,
                Resizable = true,
                Document = "~/App_Data/ExampleFiles/Default.pdf",
                Hidden = true,
                DisplayMode = DisplayMode.Window,
                WindowOptions =
                {
                    Title = "DocumentViewer as a modal dialog",
                    Modal = true,
                    Maximizable = true,
                    Minimizable = true
                }
            };

            var documentViewer3 = new DocumentViewer
            {
                Id = "documentViewer3",
                Width = 800,
                Height = 600,
                Resizable = true,
                Document = "~/App_Data/ExampleFiles/Default.pdf",
                Hidden = true,
                DisplayMode = DisplayMode.Panel,
                PanelOptions =
                {
                    Title = "DocumentViewer as a panel",
                    Collapsible = true
                }
            };

            return View(new[] {documentViewer1, documentViewer2, documentViewer3});
        }
    }
}
